qcodes/instrument_drivers/Keysight/private/Keysight_344xxA_submodules.py

```diff
@@ -172,12 +173,17 @@ def __init__(self, parent: '_Keysight_344xxA', name: str, **kwargs):
             option) or 2,000,000 readings (with the MEM option)"""))
 
         if self.parent.has_DIG:
+            if self.parent.has_MEM:
+                _max_pretrig_count = 2e6
+            else:
+                _max_pretrig_count = 5e4
+
```

Comment:
The way that I read the above comment in the count parameter is that you have a total of 50_000 or 2_000_000 points if you use pretrigger not that amount of points in the pretrigger but I have not tested this my self

Reply:
This is correct. From the manual:
![afbeelding](https://user-images.githubusercontent.com/9569658/69808473-d82a7480-11e7-11ea-888b-775dc28d8396.png)

Reply:
The way I read that thing from the manual is that the maximum for count is 50_000 or 2_000_000 if pretrigger is enabled at all. This does not say anything about the maximum number of pretrigger samples (which I agree must be the same or lower)

Reply:
ok, we read the manual again, and here's the outcome:
- the maximum allowed number of pretrigger samples ('pretrigger_count') is always equal to the current sample count ('sample.count') minus 1.
- the maximum allowed number of sample counts ('sample.count') depends on a model, on whether pretrigger samples are used or not, and the MEM option availability.
  - for the models in question (those which can have DIG), if pretriggerring is used, the maximum allowed sample count in 50k without MEM option, and 2000k with MEM option.

This last sub-bullet point is not implemented for the 'sample.count' parameter. I think its out of scope of this PR. As for the maximums for the pretrigger count, i think we can keep them as is for now, and in the bright future properly implement the fact that with the change of 'sample.count' the validator for the pretrigger_count changes as well.

@jenshnielsen would you agree?

Reply:
Yes that sounds good. 
Can we:

* move the -1 into the definition of the _max_pretrig_count variable so the name is true. 
* Add the fact that max_pretrigger_count is max_count -1 to the docs of pretrigger_count with a note to read the docs of count?

Reply:
yes! i'll do both right now!